lang/cs/Org.Apache.REEF.Client/YARN/WindowsYarnJobCommandBuilder.cs

```diff
@@ -0,0 +1,108 @@
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Text;
+using Org.Apache.REEF.Client.Yarn;
+using Org.Apache.REEF.Client.YARN.Parameters;
+using Org.Apache.REEF.Common.Files;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Client.YARN
+{
+    /// <summary>
+    /// WindowsYarnJobCommandBuilder is .NET implementation of `org.apache.reef.runtime.common.launch.JavaLaunchCommandBuilder`
+    /// This class provides the command to be submitted to RM for execution of .NET driver running on Windows environment.
+    /// </summary>
+    internal sealed class WindowsYarnJobCommandBuilder : IYarnJobCommandBuilder
+    {
+        private const int DefaultDriverMaxMemoryAllocationPoolSizeMB = 512;
+        private const int DefaultDriverMaxPermSizeMB = 128;
+        private static readonly string JavaExe = @"%JAVA_HOME%/bin/java";
+
+        private static readonly string JvmOptionsPermSize = @"-XX:PermSize=128m";
+        private static readonly string JvmOptionsMaxPermSizeFormat = @"-XX:MaxPermSize={0}m";
+        private static readonly string JvmOptionsMaxMemoryAllocationPoolSizeFormat = @"-Xmx{0}m";
+        private static readonly string ClassPathToken = @"-classpath";
+
+        private static readonly string ProcReefProperty = @"-Dproc_reef";
+        private static readonly string JavaLoggingProperty =
+            @"-Djava.util.logging.config.class=org.apache.reef.util.logging.Config";
+        private static readonly string LauncherClassName = @"org.apache.reef.bridge.client.YarnBootstrapREEFLauncher";
+        private readonly REEFFileNames _fileNames;
+        private readonly bool _enableDebugLogging;
+        private readonly IYarnCommandLineEnvironment _yarnCommandLineEnvironment;
+        private int _driverMaxMemoryAllocationPoolSizeMB;
+        private int _driverMaxPermSizeMB;
+
+
+        [Inject]
+        private WindowsYarnJobCommandBuilder(
+            [Parameter(typeof(EnableDebugLogging))] bool enableDebugLogging,
+            IYarnCommandLineEnvironment yarnCommandLineEnvironment,
+            REEFFileNames fileNames)
+        {
+            _yarnCommandLineEnvironment = yarnCommandLineEnvironment;
+            _enableDebugLogging = enableDebugLogging;
+            _fileNames = fileNames;
+            _driverMaxMemoryAllocationPoolSizeMB = DefaultDriverMaxMemoryAllocationPoolSizeMB;
+            _driverMaxPermSizeMB = DefaultDriverMaxPermSizeMB;
+        }
+
+        public IYarnJobCommandBuilder SetMaxDriverAllocationPoolSizeMB(int sizeMB)
```

Comment:
We can make them injectable and allow to set if we would allow to set later. Otherwise, just injectable.